File: skill_sdk/tests/fixtures/js-project/src/core/DefaultProcessor.js
import { DataProcessor } from './DataProcessor.js';
import { TransformData } from './TransformData.js';

/**
 * Primary implementation of DataProcessor. Mirrors DefaultProcessor in the Java fixture.
 *
 * @implements {DataProcessor}
 */
export class DefaultProcessor extends DataProcessor {
  /**
   * @param {{ getTimeout(): number; getRetries(): number }} config
   */
  constructor(config) {
    super();
    /** @type {{ getTimeout(): number; getRetries(): number }} */
    this._config = config;
  }

  /**
   * @returns {{ getTimeout(): number; getRetries(): number }}
   */
  getConfig() {
    return this._config;
  }

  /**
   * @param {string} data
   * @returns {string}
   */
  process(data) {
    if (!this.validate(data)) {
      throw new Error(`invalid data: ${data}`);
    }
    return TransformData.transform(data);
  }

  /**
   * @param {string} data
   * @returns {boolean}
   */
  validate(data) {
    return data !== null && data !== undefined && data.length > 0 && data.length < 1024;
  }
}

File: skill_sdk/tests/fixtures/java-project/src/main/java/com/fixture/core/DefaultProcessor.java
package com.fixture.core;

/**
 * Primary implementation of DataProcessor. Mirrors DefaultProcessor in main.go.
 */
public class DefaultProcessor implements DataProcessor {

    private final ProcessorConfig config;

    public DefaultProcessor(ProcessorConfig config) {
        this.config = config;
    }

    public ProcessorConfig getConfig() {
        return config;
    }

    @Override
    public String process(String data) {
        if (!validate(data)) {
            throw new IllegalArgumentException("invalid data: " + data);
        }
        return TransformData.transform(data);
    }

    @Override
    public boolean validate(String data) {
        return data != null && data.length() > 0 && data.length() < 1024;
    }
}
